superset-frontend/packages/superset-ui-chart-controls/src/shared-controls/constants.tsx

```diff
@@ -51,3 +52,29 @@ export const xAxisControlConfig = {
   },
   default: undefined,
 };
+
+export const temporalXAxisMixin = {
```

Comment:
As this is a mixin, we maybe the type could be narrowed down to something like `Pick[BaseControlConfig, 'label' | 'mapStateToProps']`?

Also, I'm wondering if we should make this a function `getXAXisControlMixin` that takes an optional argument `dataTypes: GenericDataType[]`, which if defined, would only return the requested column types? For instance, at some point in the future I'm assuming we want to support any ordinal or linear datatypes in this chart type, and in that case we would likely want to support `GenericDataType.NUMERIC` and `GenericDataType.TEMPORAL`.

Reply:
Thanks for the review and very useful case for this mixin. I changed this variable from the `temporalXAxisMixin` to the `temporalColumnMixin` and removed `mapStateToProps` from the original one from `dndGranularitySqlaControl`. I totally agree with making a function for returning specific column types from the original datasource. I will refactor this part in separate PR.